### DISCUSSION TITLE
Help needed with initial setup - The RDP connection failed to connect or run.

### DISCUSSION THREAD

**Discussion by LaPeste** [winapps-org/winapps#75] [Support]
Hi everyone,

Thank you for the amazing project.
I did my best to follow the guide but running `./installer.sh` fails. I hope this is a good place to ask for help.

I'll post as much info as possible for you to tell me what I've done wrong:

* 
    ```bash
    $ groups ${whoami}
    andrea cdrom floppy sudo audio dip video plugdev kvm netdev bluetooth lpadmin scanner libvirt
    ```
* `/etc/libvirt/qemu.conf`
    ```bash
    # The user for QEMU processes run by the system instance. It can be
    # specified as a user name or as a user id. The qemu driver will try to
    # parse this value first as a name and then, if the name doesn't exist,
    # as a user id.
    #
    # Since a sequence of digits is a valid user name, a leading plus sign
    # can be used to ensure that a user id will not be interpreted as a user
    # name.
    #
    # Some examples of valid values are:
    #
    #       user = "qemu"   # A user named "qemu"
    #       user = "+0"     # Super user (uid=0)
    #       user = "100"    # A user named "100" or a user with uid=100
    #
    user = "andrea"
    
    # The group for QEMU processes run by the system instance. It can be
    # specified in a similar way to user.
    group = "libvirt-qemu"    
  ```
* [qemu xml setup](https://pastebin.com/n7wXJwCN)
*
  ```bash
  ls -l /mnt/LinuxGames/winVM/
  total 59042452
  -rw-r--r-- 1 andrea libvirt-qemu   627519488 Jan 23 14:45 virtio-win-0.1.240.iso
  -rw-r--r-- 1 andrea libvirt-qemu  6135633920 Jan 23 14:47 Win10_22H2_EnglishInternational_x64v1.iso
  -rw------- 1 andrea libvirt-qemu 53695545344 Jan 28 12:09 win10.qcow2
  ```
* 
  ```bash
  $ virsh list
  Id   Name         State
  ----------------------------
  2    RDPWindows   running
  ```
* [output](https://pastebin.com/ewrhtxf5) of `bin/winapps check`
* 
  ```bash
  $ ./installer.sh
  ? Would you like to install for the current user or the whole system? User
  Removing any old configurations...
  grep: /home/andrea/.local/bin/*: No such file or directory
  Installing...
    Checking for installed apps in RDP machine (this may take a while)... Finished.

  The RDP connection failed to connect or run. Please confirm FreeRDP can connect with:
    bin/winapps check

  If it cannot connect, this is most likely due to:
    - You need to accept the security cert the first time you connect (with 'check')
    - Not enabling RDP in the Windows VM
    - Not being able to connect to the IP of the VM
    - Incorrect user credentials in winapps.conf
    - Not merging install/RDPApps.reg into the VM
  ```

If there is more info needed that can help debug the issue don't hesitate to ask.
Thanks to everyone for the help.

**Comment by LDprg** [comment]
Did you enable rdp and loaded the reg file? Seems like you missed that stuff (see install doc).

Also note that winapps is currently not working correctly under wayland (because of freerdp lacking support).

**Reply by LaPeste** [reply]
> Did you enable rdp and loaded the reg file?

Both are done. I just didn't have a way to show anything regarding that, beside saying: done.
If you have a way for me to share some info from the VM so you can check my registry and the likes, let me know and I'll do it right away.

And about wayland, I've read the lack of support so I went with x.org and it still did not work.

**Reply by LDprg** [reply]
Did you configure winapps with the config file (see docs)?

Maybe you want to try to connect directly with freerdp to the vm.

**Reply by LaPeste** [reply]
> Did you configure winapps with the config file (see docs)?

Yes I did that too.
I have had created `~/.config/winapps/winapps.conf` and it looks like

```ini
RDP_USER="WIN_VM_USER"
RDP_PASS="WIN_VM_PASS"
#RDP_DOMAIN="MYDOMAIN"
#RDP_IP="192.168.123.111"
RDP_SCALE=125
#RDP_FLAGS=""
#MULTIMON="true"
#DEBUG="true"
```

(I left the comments in the file for reference.)

> Maybe you want to try to connect directly with freerdp to the vm.

What are you trying to say? Not to use winapps but freeRDP directly? Why's that?

**Reply by LDprg** [reply]
The usage of freerdp directly (calling freerdp feom the console) allows you to specify whether the problem is the freerdp connection or the winapps config.

Maybe you want to specify the ip in the config manually first. However check if you are able to ping the vm, it is very likely that the vm network adapter is misconfigured.

**Reply by KernelGhost** [reply]
There is an unfortunate issue with your configuration file. `RDP_SCALE` only supports values of `100`, `140` and `180`. Values other than these previously caused `FreeRDP` commands to mysteriously fail. You can read #145 for more information. The repository has been updated to automatically correct incompatible scaling values to the nearest supported integer. If you'd like, you can try pulling the latest changes from the repository and trying again.
